lib/bolt/cli.rb

```diff
@@ -559,6 +563,40 @@ def initialize_project
       ok ? 0 : 1
     end
 
+    def migrate_project
+      if inventory.version == 2
+        ok = true
+      else
+        inventoryfile = config.inventoryfile || config.boltdir.inventory_file
+        inv = YAML.load_file(inventoryfile)
+
+        inv = Bolt::Util.walk_pairs(inv) do |k, v|
```

Comment:
I think you can just use `Bolt::Util#walk_keys`. 
```
        inv = Bolt::Util.walk_keys(inv) do |k|
          case k
          when k == 'nodes'
            k = 'target'
          when k == 'name'
            k = 'uri'
          else
            k
          end
        end
```

Reply:
That would update every name key to uri, instead of just the name key for targets. Names for groups/facts/etc would be changed as well, which we don't want .

Reply:
ahhh, yeah. forgot about that.

Reply:
I guess the reason I brought that up was more that I just don't think we need a util method that walks pairs. I think we can do it with an existing one. What about something like:
```ruby
        inv = Bolt::Util.walk_vals(inv) do |v|
          if v.is_a?(Hash) && (grps = v['groups'])
            grps.map do |grp|
              grp.map do |key, val|
                if key == 'nodes'
                  key == 'targets'
                  val = val.map do |target|
                    target['uri'] = target.delete('name') if target.is_a?(Hash)
                    target
                  end
                end
              end
            end
          end
          v
        end
```
it is nasty but I think we could make it better maybe?

Reply:
I had something similar to that prior to opening the PR, but also felt it was messy. I don't really have a preference for one approach v another, so if we want to remove `walk_pairs` I can revert back to what I had previously.

Reply:
@donoghuc Is your concern adding more functions to Util, or overcomplicating this traversal since we can filter what we walk here based on `Groups`?  When adding `walk_vals` we only used it in one place, so I don't see a problem with adding a function to `walk_hashes` and it could potentially be useful elsewhere where context for traversing objects is necessary (like it is here). 

If the concern is wanting to filter the top-level keys we traverse, I think we could write a more fine-tuned traversal method and include it here instead. Given the algorithm above I don't think that's your concern?

Worth noting the above algorithm misses nodes in the top-level `nodes` key (that are part of the `all` group)

Reply:
@beechtom This seems to work:
```
        inv = Bolt::Util.walk_vals(inv) do |v| 
          if v.is_a?(Hash)
            v['targets'] = v.delete('nodes') if v.has_key?('nodes')
            if v.has_key?('targets')
              v['targets'] = v['targets'].map do |target|
                target['uri'] = target.delete('name') if target.is_a? Hash
                target
              end 
            end 
          end 
          v   
        end 
```

Reply:
It might be clearer to do this with explicit recursion than with a helper function.

```rb
def migrate_group(group)
  if group.key?('nodes')
    targets = group['nodes'].map do |target|
      target['uri'] = target.delete('name') if target.is_a?(Hash)
      target
    end
    group.delete('nodes')
    group['targets'] = targets
  end
  (group['groups'] || {}).map { |name, subgroup| migrate_group(subgroup) }
end
```

Reply:
Updated to use `walk_vals`.

Reply:
Does the recursive algo that Nick suggest work? I kind of like that one for readability.

Reply:
I'm probably not using it correctly, but I can't get it to work.

Reply:
Seems to work for me with 
```ruby
    def migrate_group(group)
      if group.key?('nodes')
        targets = group['nodes'].map do |target|
          target['uri'] = target.delete('name') if target.is_a?(Hash)
          target
        end
        group.delete('nodes')
        group['targets'] = targets
      end
      (group['groups'] || []).map do |subgroup|
        migrate_group(subgroup)
      end
    end
```
then simply replace the walk_vals logic with 
`migrate_groups(inv)`

Reply:
Updated.